src/pipelines/token_classification.rs

```diff
@@ -0,0 +1,518 @@
+}
+
+impl TokenizerOption {
+    /// Interface method to load a tokenizer from file
+    pub fn from_file(model_type: ModelType, vocab_path: &str, merges_path: Option<&str>, lower_case: bool) -> Self {
+        match model_type {
+            ModelType::Bert | ModelType::DistilBert => TokenizerOption::Bert(BertTokenizer::from_file(vocab_path,lower_case)),
+            ModelType::Roberta => TokenizerOption::Roberta(RobertaTokenizer::from_file(vocab_path, merges_path.expect("No merges specified!"), lower_case)),
+        }
+    }
+
+    /// Returns the model type
+    pub fn model_type(&self) -> ModelType {
+        match *self {
+            Self::Bert(_) => ModelType::Bert,
+            Self::Roberta(_) => ModelType::Roberta
+        }
+    }
+
+    /// Interface method
+    pub fn encode_list(&self, text_list: Vec<&str>, max_len: usize, truncation_strategy: &TruncationStrategy, stride: usize) -> Vec<TokenizedInput> {
+        match *self {
+            Self::Bert(ref tokenizer) => tokenizer.encode_list(text_list, max_len, truncation_strategy, stride),
+            Self::Roberta(ref tokenizer) => tokenizer.encode_list(text_list, max_len, truncation_strategy, stride)
+        }
+    }
+}
+
+
+
+impl TokenClassificationOption {
+    /// Instantiate a new token sequence classification model of the supplied type.
+    ///
+    /// # Arguments
+    ///
+    /// * `model_type` - `ModelType` indicating the model type to load (must match with the actual data to be loaded)
+    /// * `p` - `tch::nn::Path` path to the model file to load (e.g. model.ot)
+    /// * `config` - A configuration (the model type of the configuration must be compatible with the value for
+    /// `model_type`)
+    ///
+    pub fn new(model_type: ModelType, p: &tch::nn::Path, config: &ConfigOption) -> Self {
+        match model_type {
+            ModelType::Bert => {
+                if let ConfigOption::Bert(config) = config {
+                    TokenClassificationOption::Bert(BertForTokenClassification::new(p, config))
+                } else {
+                    panic!("You can only supply a BertConfig for Bert!");
+                }
+            },
+            ModelType::DistilBert => {
+                if let ConfigOption::DistilBert(config) = config {
+                    TokenClassificationOption::DistilBert(DistilBertForTokenClassification::new(p, config))
+                } else {
+                    panic!("You can only supply a DistilBertConfig for DistilBert!");
+                }
+            }
+            ModelType::Roberta => {
+                if let ConfigOption::Bert(config) = config {
+                    TokenClassificationOption::Roberta(RobertaForTokenClassification::new(p, config))
+                } else {
+                    panic!("You can only supply a BertConfig for Roberta!");
+                }
+            }
+        }
+    }
+
+    /// Returns the `ModelType` for this TokenClassificationOption
+    pub fn model_type(&self) -> ModelType {
```

Comment:
I am not sure if this is used somewhere? I may have missed it

Reply:
You may be right yeah, I usually do matching on the enums directly so this might not be used yet. I'm not sure if it might come in handy still at some point (facilitates easy comparisons and compatibility checks between the different ``*Options``).